src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,336 @@
+     */
+    public String execute(TaskList tasks, Storage storage) throws NoSuchCommandException, InvalidIndexException,
+            UnmatchedArgumentException, EmptyDescriptionException {
+
+        String result = "";
+        if (command.equals("list")) {
+            result = tasks.listAllTask();
+        } else if (command.equals("mark")) {
+
+            if (!isValidDetail) {
+                throw new InvalidIndexException();
+            }
+            int index = Integer.parseInt(details);
+            if (index > 0 && index <= tasks.size()) {
+                result = mark(index, tasks, storage);
+            } else {
+                throw new InvalidIndexException();
+            }
+        } else if (command.equals("delete")) {
+
+            if (!isValidDetail) {
+                throw new InvalidIndexException();
+            }
+            int index = Integer.parseInt(details);
+            if (index > 0 && index <= tasks.size()) {
+                result = delete(index, tasks, storage);
+            } else {
+                throw new InvalidIndexException();
+            }
+        } else if (command.equals("unmark")) {
+
+            if (!isValidDetail) {
+                throw new InvalidIndexException();
+            }
+            int index = Integer.parseInt(details);
+            if (index > 0 && index <= tasks.size()) {
+                result = unmark(index, tasks, storage);
+            } else {
+                throw new InvalidIndexException();
+            }
+        } else if (command.equals("todo")) {
+            result = addTodo(details, tasks, storage);
+        } else if (command.equals("deadline")) {
+            result = addDeadline(details, tasks, storage);
+        } else if (command.equals("event")) {
+            result = addEvent(details, tasks, storage);
+        } else if (command.equals("due")) {
+            result = checkTaskDue(details, tasks);
+        } else if (command.equals("find")) {
+            result = find(details, tasks);
+        } else if (command.equals("bye")) {
+            isExit();
+        } else {
+            throw new NoSuchCommandException();
+        }
+        return result;
+    }
+
+    /**
+     * Marks a task as done and updates it into the storage (file).
+     *
+     * @param i The index of the task to be marked as done in the list.
+     * @param tasks The task list containing the tasks.
+     * @param storage The storage object that contains the file for updating.
+     * @return The response of the bot after mark the task as done.
+     */
+    public String mark(int i, TaskList tasks, Storage storage) {
+
+        String result = "";
+        tasks.get(i - 1).mark();
+        result += Ui.showLine() + "\n" + " Nice! I've marked this task as done:" + "\n" + " " + tasks.get(i - 1).toString();
+        result += "\n";
+        result += Ui.showLine();
+        storage.writeInto(tasks);
+        return result;
+    }
```

Comment:
I see you mark method here! Why don't you let this method call parse the details and settle the exceptions? (extension from prev comment)